### PR TITLE
create separated page in functional programming

### LINKED ISSUE #141: Make the display of a new chapter more clear to the reader
> when walking the chapters using the next button, I initially thought the Functional Programming chapter was part of the anti-patterns chapter and was confused why FP was an anti-pattern.

src.: [reddit](https://www.reddit.com/r/rust/comments/kowtqn/rust_design_patterns_now_also_as_a_book/ghvvj0s/?utm_source=reddit&utm_medium=web2x&context=3)

### PR DESCRIPTION
**Pull request by MarcoIeni**
fix #141

### CODE REVIEW

File: functional/declarative.md
Comment (simonsan) [inline functional/declarative.md:1]: Would call it `fp-paradigms.md` and the headline `Function Programming paradigms` as well. In the `SUMMARY` I would call it just `Paradigms` as it's already a subsection of FP.

Comment (simonsan) [reply functional/declarative.md:1]: thoughts? @MarcoIeni

Comment (MarcoIeni) [reply functional/declarative.md:1]: if you do it like that it looks like "imperative" is a fp paradigm

Comment (MarcoIeni) [reply functional/declarative.md:1]: Anyway, I agree that current names are not that good

Comment (simonsan) [reply functional/declarative.md:1]: Ah read it yesterday and forgot to answer, sorry. What would be your proposal?

Comment (MarcoIeni) [reply functional/declarative.md:1]: The headline and summary title could be `Declarive paradigm`?

Comment (pickfire) [reply functional/declarative.md:1]: No, this should be called paradigms. If we put `Declarative paradigm` here and we talk about imperative, it is a bit confusing, so better just put the general world that covers both.

Comment (MarcoIeni) [reply functional/declarative.md:1]: Ok, should we go with `Programming paradigms`? We can also link [this](https://en.wikipedia.org/wiki/Programming_paradigm) in the introduction

Comment (MarcoIeni) [reply functional/declarative.md]: simonsan, tell me if it looks ok now :)

File: SUMMARY.md
Comment (simonsan) [inline SUMMARY.md:43]:
```suggestion
  - [Programming paradigms](./functional/paradigms.md)
```

File: functional/index.md
Comment (simonsan) [inline functional/index.md:4]: I think this intro should link at least also to the content of the folder itself, like a TOC, in this case just `./paradigms.md`.

Also it should explain what this folder will contain, so i feel this sentence is a bit short? Maybe we write some basic introductory sentences as in the other `./**/intro.md`s for that corresponding subtopics?

File: functional/declarative.md
Comment (simonsan) [reply functional/declarative.md]: reviewed it

File: SUMMARY.md
Comment (MarcoIeni) [reply SUMMARY.md:43]: Maybe better "Programming paradigms" as the page title?

File: functional/index.md
Comment (MarcoIeni) [reply functional/index.md:4]: > I think this intro should link at least also to the content of the folder itself

I don't like it, the toc is already on the left. The rust book doesn't do that for example.

> Also it should explain what this folder will contain

Yes, a better intro is missing.

Comment (simonsan) [reply functional/index.md:4]: I thought we did it for the top layer as well, but indeed, we did it only for the second layer (FFI) so yeah `graphs` would be a candidate then for a TOC. my bad :)

File: SUMMARY.md
Comment (simonsan) [reply SUMMARY.md:43]: Yeah, would be clearer, I thought that it looks repeated, because over it will be `Funtional Programming` and then the next line starts with `Programming Paradigms` again. Like Programming ... Programming.

Comment (MarcoIeni) [reply SUMMARY.md]: done in 0e89dacb7

File: functional/index.md
Comment (MarcoIeni) [reply functional/index.md:4]: I added an introduction (by copy pasting from wikipedia ^^'). Tell me if it is ok now.

File: functional/paradigms.md
Comment (pickfire) [inline functional/paradigms.md:41]: I think we should mention that one can just `(1..11).sum()`.
```suggestion
println!("{}", (1..11).fold(0, |a, b| a + b)); // `.sum()` is a better alternative, this is just for illustration
```

File: functional/paradigms.md
Comment (pickfire) [inline functional/paradigms.md:49]:
```suggestion
Here, we are composing functions of addition (this closure: `|a, b| a + b`) with a range from 1 to 10.
```
Extra closing bracket.

File: functional/paradigms.md
Comment (pickfire) [inline functional/paradigms.md:50]: Even though it is being moved. I think it is better to use `acc` and `x` to follow the standard library docs and being easier to read rather than `a` and `b`.

**Review by pickfire — APPROVED** [review]
Other than the mentioned comments other parts looks good to me, even though the mentioned comments are unrelated.

**Comment by MarcoIeni** [comment]
Thank you for reviewing this, pickfire.
I opened #207 to track the changes you proposed, I hope you don't mind, but it's better to keep PRs small and on focus.
I like your proposals, we can work on them once simonsan merges this. (I can't merge this because I opened it 😅).

**Review by pickfire — APPROVED** [review]
Looks good to me then since it does what this PR mentioned. Heh, @MarcoIeni you are stuck in an issue similar to me, luckily the other reviewer is fast, in the other project the other reviewers are not so fast.

**Review by simonsan — APPROVED** [review]

**Pull request merged**
